### PR TITLE
fix(typos): add github action to prevent typos + fix typos

### PR DESCRIPTION
**Pull request by stavares843**
## What kind of change does this PR introduce?

This introduces a Github Action that automatically scans the whole codebase for typos and automatically adds a Pull Request suggesting those code changes.

## What is the current behavior?

No CI to prevent typos.

## What is the new behavior?

With a Github Action to prevent typos, we can avoid typos in the codebase which results in saving time updating either code or docs with typos.

## Additional context

I got to know this project in Hacker News, so I though I could add this contribution 🔨

### CODE REVIEW

**Review by MildTomato — CHANGES_REQUESTED** [review]

**Review by MildTomato — APPROVED** [review]

**Review by MildTomato — DISMISSED** [review]
sorry, accidentally approved

**Review by kiwicopple — APPROVED** [review]

**Comment by MildTomato** [comment]
@stavares843


ok to fix the build error in the studio ?

**Comment by MildTomato** [comment]
@stavares843

Screenshot 2022-10-19 at 3 46 55 PM
ok to fix the build error in the studio ?

**Review by MildTomato — CHANGES_REQUESTED** [review]
just to fix the build error :D 🚀 

Looks like a component misnamed

**Comment by stavares843** [comment]
thanks for reviewing! will check 🔨

**Comment by stavares843** [comment]
addressed 🔨

File: i18n/README.fr.md [hunk 1]
```diff
@@ -66,7 +66,7 @@ Vous pouvez aussi [auto-héberger](https://supabase.com/docs/guides/hosting/over
 
 #### Bibliothèques clients
 
-Notre bibliothèque client est modulaire. Chaque sous-bibliothèque est une implémentation autonome pour un seul système externe. C'est l'une des façons dont nous soutenons les outils existants.
```
[the remark this answers is not in the document]
Comment (stavares843) [reply i18n/README.fr.md]: reverted this changed and added a `.misspell-fixer.ignore` file, where i added the `i18n` to be whitelisted, will test more to make sure it works 🔨

**Comment by kiwicopple** [comment]
this is an epic contribution 🔥 

@MildTomato - you OK to merge? You have some requested changes

**Comment by stavares843** [comment]
thanks so much @kiwicopple 🎉 This is a great project! I noticed some open positions, I applied for the Support Engineer position 📺

**Comment by MildTomato** [comment]
@stavares843 still have some builds failing.
Is it building locally?

Screenshot 2022-10-24 at 7 06 16 PM

ps. this is great btw 🔥 nice work

**Comment by stavares843** [comment]
Captura de ecrã 2022-10-24, às 13 40 09

i also double checked CI, and the latest one is green 🎉 

that one you posted, seems a previous one that was failing 🔨

**Comment by MildTomato** [comment]
@stavares843 ahh sorry about that, i was looking at wrong thing 🙈 
running checks again now 🚀 !

**Comment by stavares843** [comment]
no worries, thanks for running the CI again! 🔨 all green now 🎉

**Review by MildTomato — APPROVED** [review]

**Comment by MildTomato** [comment]
Just getting a supabase team FE member to review as well, just to double check.
lgtm though 🚀 

thanks @stavares843 - great contribution !

**Review by pucho — APPROVED** [review]
This looks great, thanks for the contribution!

**Comment by stavares843** [comment]
thanks 🎉 

here it is, first PR added by the bot 🔨 


https://github.com/supabase/supabase/pull/9916

**Pull request merged**